### PR TITLE
Improved Linux SerialPort lookup

### PR DESCRIPTION
**Pull request by bart-jansen**
## Description
Improved Linux SerialPort lookup to have better support for devices connected on /dev/ttyUSB, /dev/ttyACM and /dev/ttyAMA.

## Motivation and Context
When trying to enumerate the devices on Raspbian, the device was available under /dev/ttyUSB0 but previously these ports were never scanned because it was only scanning /dev/ttyGS and /dev/ttyS.

## Types of changes

- [X] Improvement (non-breaking change that improves a feature, code or algorithm)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by Ellerbach — APPROVED** [review]
Thanks for the improvement and testing all this with various devices!

**Comment by Ellerbach** [comment]
@bart-jansen please take care of the SonarCloud code smells, push a fix and we should be good to merge.

File: nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs [hunk 1]
```diff
@@ -99,63 +99,23 @@ public List<string> GetPortNames()
...
         {
-            const string sysTtyDir = "/sys/class/tty";
-            const string sysUsbDir = "/sys/bus/usb-serial/devices/";
-            const string devDir = "/dev/";
+            List<string> ports = new List<string>();
 
-            if (Directory.Exists(sysTtyDir))
+            string[] ttys = System.IO.Directory.GetFiles("/dev/", "tty*");
+            foreach (string dev in ttys)
             {
```
Comment (CarlosSardo) [inline nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs:106]: I believe you could simplify it with a LINQ expression as it follows:

`
            var ports = matches.Where(dev =>
                                ttys.All(m => m.StartsWith(dev, StringComparison.InvariantCultureIgnoreCase)))
                            .ToList();`

**Review by CarlosSardo — APPROVED** [review]

File: nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs (hunk 1 above)
Comment (Ellerbach) [reply nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs:106]: not super sure Linq does simplify anything :-D Feels like it brings more complexity on how you can read the code.

**Pull request merged**